src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -69,7 +90,10 @@ private BoundExpression LowerSwitchExpression(BoundConvertedSwitchExpression nod
                     var sectionBuilder = ArrayBuilder<BoundStatement>.GetInstance();
                     sectionBuilder.AddRange(switchSections[arm.Syntax]);
                     sectionBuilder.Add(_factory.Label(arm.Label));
-                    sectionBuilder.Add(_factory.Assignment(_factory.Local(resultTemp), _localRewriter.VisitExpression(arm.Value)));
+                    var loweredValue = _localRewriter.VisitExpression(arm.Value);
+                    if (GenerateSequencePoints)
+                        loweredValue = new BoundSequencePointExpression(arm.Value.Syntax, loweredValue, loweredValue.Type);
```

Comment:
I thought we now avoid directly generating sequence points (except for hidden sequence points). 

I expected that sequence point creation should be delegated to the instrumenter (eg. `_localRewriter._instrumenter.InstrumentXYZ();` kind of thing).
Or is this intentional (ie. not a scenario relevant to general instrumentation or LUT analysis)?

Reply:
I plan to follow up with instrumentation requirements and interface entry points separately.  There are three issues I'm tracking for that:
- Instrumentation for the when clause
- Instrumentation for the switch expression arm's expression
- What to do about instrumentation for all the other places where we generate sequence points that are probably not instrumentable (e.g. line 75)
See [https://github.com/dotnet/roslyn/issues/42810](https://github.com/dotnet/roslyn/issues/42810)

---
In reply to: [398869263](https://github.com/dotnet/roslyn/pull/42711#discussion_r398869263) [](ancestors = 398869263)

Reply:
Cool. Consider leaving a link here to minimize hunting of places that needs to be hooked up with instrumenter.